vmm/src/vm.rs

```diff
@@ -2026,7 +2026,20 @@ impl Vm {
 
         // Do earlier to parallelise with loading kernel
         #[cfg(target_arch = "x86_64")]
-        let rsdp_addr = self.create_acpi_tables();
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "sev_snp")] {
+                let rsdp_addr = if self.sev_snp_enabled {
```

Comment:
You don't need to add this flag to the `Vm` struct, as you can query `self.config.lock().unwrap().is_sev_snp_enabled();`.